packages/aws-cdk-lib/aws-cloudfront/lib/experimental/edge-function.ts

```diff
@@ -199,9 +197,7 @@ export class EdgeFunction extends Resource implements lambda.IVersion {
     });
 
     const resourceType = 'Custom::CrossRegionStringParameterReader';
-    const serviceToken = CustomResourceProvider.getOrCreate(this, resourceType, {
-      codeDirectory: path.join(__dirname, '..', '..', '..', 'custom-resource-handlers', 'dist', 'aws-cloudfront', 'edge-function'),
-      runtime: CustomResourceProviderRuntime.NODEJS_18_X,
+    const serviceToken = CrossRegionStringParamReaderProvider.getOrCreate(this, resourceType, {
```

Comment:
Why 

```ts
const requestorFunction = new CertificateRequestCertificateRequestProvider(this, 'CertificateRequestorFunction', {
```

For the previous CR, but `getOrCreate` for this one?

Wouldn't it make sense to have the same API all the way around?